app/components/files/audio_file.tsx

```diff
@@ -0,0 +1,218 @@
+        padding: 12,
+        overflow: 'hidden',
+        flexDirection: 'row',
+        gap: 12,
+        alignItems: 'center',
+    },
+    playButton: {
+        flexDirection: 'column',
+        justifyContent: 'center',
+        alignItems: 'center',
+        backgroundColor: changeOpacity(theme.buttonBg, 0.12),
+        borderRadius: 100,
+        width: 40,
+        height: 40,
+    },
+    playIcon: {
+        color: theme.buttonBg,
+    },
+    downloadIcon: {
+        color: changeOpacity(theme.centerChannelColor, 0.56),
+    },
+    progressBar: {
+        flex: 1,
+    },
+    timerText: {
+        color: theme.centerChannelColor,
+        ...typography('Body', 75, 'SemiBold'),
+    },
+}));
+
+const AudioFile = ({file, canDownloadFiles}: Props) => {
+    const intl = useIntl();
+    const theme = useTheme();
+    const style = getStyleSheet(theme);
+    const [hasPaused, setHasPaused] = useState<boolean>(true);
+    const [hasError, setHasError] = useState<boolean>(false);
+    const [hasEnded, setHasEnded] = useState<boolean>(false);
+    const [progress, setProgress] = useState<number>(0);
+    const [timeInMinutes, setTimeInMinutes] = useState<string>('0:00');
+    const [duration, setDuration] = useState<number>(0);
+    const videoRef = useRef<VideoRef>(null);
+
+    useEffect(() => {
+        if (hasEnded) {
+            const timer = setTimeout(() => {
+                setHasPaused(true);
+                setProgress(0);
+                setHasEnded(false);
+            }, 100);
+
+            return () => clearTimeout(timer);
+        }
+
+        return () => null;
+    }, [hasEnded]);
+
+    const source = useMemo(() => ({uri: file.uri}), [file.uri]);
+
+    const {toggleDownloadAndPreview} = useDownloadFileAndPreview();
+
+    const onPlayPress = () => {
+        setHasPaused(!hasPaused);
+    };
+
+    const onDownloadPress = async () => {
+        if (!canDownloadFiles) {
+            alertDownloadDocumentDisabled(intl);
+            return;
+        }
+
+        toggleDownloadAndPreview(file);
+    };
+
+    const loadTimeInMinutes = (timeInSeconds: number) => {
+        const minutes = Math.floor(timeInSeconds / 60);
+        const seconds = Math.floor(timeInSeconds % 60);
+        setTimeInMinutes(`${minutes}:${seconds.toString().padStart(2, '0')}`);
+    };
+
+    const onLoad = (loadData: OnLoadData) => {
+        loadTimeInMinutes(loadData.duration);
+        setDuration(loadData.duration);
+    };
+
+    const onProgress = (progressData: OnProgressData) => {
+        if (hasPaused) {
+            return;
+        }
+        const {currentTime, playableDuration} = progressData;
+        setProgress(currentTime / playableDuration);
+        throttle(() => loadTimeInMinutes(currentTime), 1000)();
+    };
+
+    const onEnd = () => {
+        if (videoRef.current) {
+            videoRef.current.seek(0);
+        }
+        setHasEnded(true);
+    };
+
+    const onError = () => {
+        setHasError(true);
+    };
```

Comment:
We don't get any information about the error? Or we just don't care?

Reply:
great question @larkox 

I just don't know what we could do to it, do we have some internal logging or telemetry? something like it?

Because the error is not user friendly as you can see:
![image](https://github.com/user-attachments/assets/7600bb7b-62bd-4675-b9b8-3d410e738190)

Reply:
I think you can pass the error to `logDebug` at least. Since we know it may provide a `errorString`, and we want to keep the logs a bit cleaner, we can do something like:

```
logDebug((error && typeof error === 'object' && 'errorString' in error) ? error.errorString : error);
```

Reply:
Done!! thank you!